src/ahrens_lab_to_nwb/yu_mu_cell_2019/yu_mu_cell_2019_imaging_extractor.py

```diff
@@ -15,36 +15,62 @@ class AhrensHdf5ImagingExtractor(ImagingExtractor):
     extractor_name = "AhrensHdf5Imaging"
     mode = "file"
 
-    def __init__(self, file_path: PathType, sampling_frequency: float):
+    def __init__(
+        self,
+        file_path: PathType,
+        sampling_frequency: float,
+        region: Optional[Literal["top", "bottom"]] = None,
+        shape: Optional[Tuple[int]] = None,  # If specified, don't grab from file
+        dtype: Optional[np.dtype] = None,  # If specified, don't grab from file
```

Comment:
I added these in to speed up initialization of the interface. The new sessions have up to 15,000 frames, so it was actually getting pretty expensive to initialize every single io just to grab the basic meta info on shape/dtype to check for consistency.

This does perhaps make it possible for a problematic deviation to occur but I'm willing to risk that at this point. It was taking an absurd amount of time to initialize the interface/MultiImagingExtractor otherwise.

Reply:
Interesting.